tests/integration/models/torch/lighting_modules/test_segmentation.py

```diff
@@ -0,0 +1,159 @@
+    metric_name = "BinaryF1Score"
+    print(training_metric_values)
+    # Check if the chosen metric is improving
+    for i in range(1, len(training_metric_values)):
+        assert (
+            training_metric_values[i][metric_name]
+            > training_metric_values[i - 1][metric_name]
+        )
+
+
+def test_testing_without_checkpoint(segmentation_module: LightningModule):
+    trainer = Trainer(max_epochs=1, accelerator="gpu", devices=1)
+    dataset = DummyDataset(num_samples=100)
+    dataloader = DataLoader(dataset, batch_size=4)
+    test_results = trainer.test(segmentation_module, dataloaders=dataloader)
+
+    # Check if test_results is a non-empty list
+    assert len(test_results) > 0
+
+    # Check if the first dictionary in test_results contains the expected metrics
+    expected_metrics = {
+        "test_BinaryF1Score",
+        "test_BinaryJaccardIndex",
+        "test_BinaryPrecision",
+        "test_BinaryRecall",
+    }
+    for metric in expected_metrics:
+        assert metric in test_results[0]
+
+
+def test_predicting_without_checkpoint(segmentation_module: LightningModule):
+    trainer = Trainer(max_epochs=1, devices=1)
+    predict_dataloader = DataLoader(DummyDataset(10), batch_size=4)
+    # predictions = predict(segmentation_module, predict_dataloader)
+    trainer.predict(segmentation_module, dataloaders=predict_dataloader)
+
+
+def test_testing_with_checkpoint(segmentation_module: LightningModule):
+    checkpoint_path = "checkpoints"
+    os.makedirs(checkpoint_path, exist_ok=True)
+
+    # # Set up the trainer
+    # checkpoint_callback = ModelCheckpoint(dirpath=checkpoint_path, save_top_k=-1)
+    trainer = Trainer(
+        max_epochs=2, devices=1, default_root_dir=checkpoint_path
+    )
+
+    dataset = DummyDataset(num_samples=100)
+    dataloader = DataLoader(dataset, batch_size=4)
+
+    # Fit the model
+    trainer.fit(segmentation_module, dataloader)
+
+    # Test the loaded model
+    trainer.test(
+        segmentation_module,
+        ckpt_path="checkpoints/lightning_logs/version_0/checkpoints/epoch=1-step=50.ckpt",
+        dataloaders=dataloader,
+    )
+
+    # Delete the checkpoint folder
+    shutil.rmtree(checkpoint_path)
+
+
+def test_predicting_with_checkpoint(segmentation_module: LightningModule):
+    checkpoint_path = "checkpoints"
+
+    # Instantiate the DummyDataModule
+    data_module = DummyDataModule(num_samples=100, batch_size=4)
+
+    # Instantiate the Trainer with desired configurations
+    trainer = Trainer(
+        max_epochs=2, devices=1, default_root_dir=checkpoint_path
+    )
+
+    # Fit the model (train and validate)
+    trainer.fit(segmentation_module, data_module)
+
+    # Create a DataLoader for the prediction data
+    predict_dataloader = DataLoader(DummyDataset(10), batch_size=4)
+
+    trainer.predict(
+        segmentation_module,
+        ckpt_path="checkpoints/lightning_logs/version_0/checkpoints/epoch=1-step=50.ckpt",
```

Comment:
seems like you're training a model twice to obtain similar checkpoints. I would suggest using fixture with scope session:
def get_model_ckpt(tmp_path):
    temp_dir = tmpdir_factory.mktemp('temp') 
    # use temp_dir  in trainer ...
    trainer.fit(model, datamodule)
    yield temp_dir / "checkpoints/lightning_logs/version_0/checkpoints/epoch=1-step=50.ckpt"
    

For more resources: 
https://habr.com/ru/articles/448786/
https://betterprogramming.pub/understand-5-scopes-of-pytest-fixtures-1b607b5c19ed?gi=487ecefbaf64